packages/jsii-rosetta/lib/rosetta-translator.ts

```diff
@@ -83,14 +83,15 @@ export class RosettaTranslator {
    *
    * Will remove the cached snippets from the input array.
    */
-  public readFromCache(snippets: TypeScriptSnippet[], addToTablet = true): ReadFromCacheResults {
+  public readFromCache(snippets: TypeScriptSnippet[], addToTablet = true, compiledOnly = false): ReadFromCacheResults {
```

Comment:
I would be fine making this the default. Let's not add switches we're not going to use.

Reply:
I actually think the flag is necessary here. We pass in `options.includeCompilerDiagnostics` into `compiledOnly`. Without it, `rosetta:extract --no-compile` will not take advantage of the cache. For example:

run `rosetta:extract --no-compile`. examples added to tablet and cache.
run  `rosetta:extract --no-compile` again. we have cached examples, but they are not used because `didCompile` is undefined. So we get cache misses for all snippets.

Now I don't know why one would do this. But as long as we support `--no-compile` it feels iffy to say "btw, you won't get to use the cache anymore, haha".  

Unless you simply mean you want to default to `compiledOnly = true`?

Reply:
Holy shit it breaks my brain 😅.

I think you are right though, running through some cases and the alternatives feel suboptimal. Shipping this!